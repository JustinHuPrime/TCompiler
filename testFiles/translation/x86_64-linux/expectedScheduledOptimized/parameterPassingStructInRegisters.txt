testFiles/translation/x86_64-linux/input/parameterPassingStructInRegisters.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    OFFSET_LOAD(TEMP(temp6, 4, 12, MEM), REG(rdi, 8), CONSTANT(8, LONG(0))),
    OFFSET_LOAD(TEMP(temp6, 4, 12, MEM), REG(xmm0, 4), CONSTANT(8, LONG(8))),
    MOVE(TEMP(temp9, 8, 8, GP), GLOBAL(_T3foo3bar)),
    MOVE(TEMP(temp10, 4, 12, MEM), TEMP(temp6, 4, 12, MEM)),
    OFFSET_STORE(REG(rdi, 8), TEMP(temp10, 4, 12, MEM), CONSTANT(8, LONG(0))),
    OFFSET_STORE(REG(xmm0, 4), TEMP(temp10, 4, 12, MEM), CONSTANT(8, LONG(8))),
    CALL(TEMP(temp9, 8, 8, GP)),
    RETURN(),
  ),
)
