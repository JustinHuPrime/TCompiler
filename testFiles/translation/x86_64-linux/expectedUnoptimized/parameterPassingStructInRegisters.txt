testFiles/translation/x86_64-linux/input/parameterPassingStructInRegisters.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    OFFSET_LOAD(TEMP(temp6, 4, 12, MEM), REG(rdi, 8), CONSTANT(8, LONG(0))),
    OFFSET_LOAD(TEMP(temp6, 4, 12, MEM), REG(xmm0, 4), CONSTANT(8, LONG(8))),
    JUMP(CONSTANT(8, LOCAL(3))),
  ),
  BLOCK(3,
    MOVE(TEMP(temp9, 8, 8, GP), CONSTANT(8, GLOBAL(_T3foo3bar))),
    JUMP(CONSTANT(8, LOCAL(7))),
  ),
  BLOCK(7,
    MOVE(TEMP(temp10, 4, 12, MEM), TEMP(temp6, 4, 12, MEM)),
    JUMP(CONSTANT(8, LOCAL(8))),
  ),
  BLOCK(8,
    NOP(),
    OFFSET_STORE(REG(rdi, 8), TEMP(temp10, 4, 12, MEM), CONSTANT(8, LONG(0))),
    OFFSET_STORE(REG(xmm0, 4), TEMP(temp10, 4, 12, MEM), CONSTANT(8, LONG(8))),
    CALL(TEMP(temp9, 8, 8, GP)),
    JUMP(CONSTANT(8, LOCAL(4))),
  ),
  BLOCK(4,
    RETURN(),
  ),
)
