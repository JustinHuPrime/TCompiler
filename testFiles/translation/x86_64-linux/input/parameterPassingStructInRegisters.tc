module foo;

struct s {
  int i;
  float f;
  float g;
};

void bar(s a) {
  bar(a);
}