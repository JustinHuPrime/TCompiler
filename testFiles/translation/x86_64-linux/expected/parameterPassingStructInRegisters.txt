testFiles/translation/x86_64-linux/input/parameterPassingStructInRegisters.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    OFFSET_LOAD(TEMP(temp6, 4, 12, MEM), REG(rdi), OFFSET(0)),
    OFFSET_LOAD(TEMP(temp6, 4, 12, MEM), REG(xmm0), OFFSET(8)),
    JUMP(LABEL(.L3)),
  ),
  BLOCK(3,
    JUMP(LABEL(.L4)),
  ),
  BLOCK(4,
    RETURN(),
  ),
)
